docs/proposals/makefile-improvement-proposal.md

```diff
@@ -0,0 +1,68 @@
+# Proposal for Makefile improvements
+
+## Introduction
+
+The motivation for this proposal is to enable developers working on Argo Workflows to use build tools in a more reproducible way.
+Currently the Makefile is unfortunately too opinionated and as a result is often a blocker when first setting up Argo Workflows locally.
+I believe we should shrink the responsibilities of the Makefile and where possible outsource areas of responsibility to more specialized technology, such
+as Devenv/Nix in the case of dependency management.
+
+## Proposal Specifics
+
+In order to better address reproducibility, it is better to split up the duties the Makefile currently performs into various sub components, that can be assembled in more appropriate technology. One important aspect here is to completely shift the responsibility of dependency management away from the Makefile and into technology such as Nix or Devenv.
```

Comment:
I would also add a sentence here that this will improve access to the Argo Workflows open source code for non-Mac devs and make it easier to run it locally.